README.md

```diff
@@ -1,46 +1,64 @@
-# devcontainers/ci
+# Dev Container Build and Run (devcontainers/ci)
 
-devcontainers/ci contains a GitHub action and Azure DevOps task aimed at making it easier to re-use a [development container](https://github.com/devcontainers/spec) in a GitHub workflow or Azure DevOps pipeline.
+The Dev Container Build and Run GitHub Action is aimed at making it easier to re-use [Dev Containers](https://containers.dev) in a GitHub workflow. The Action supports using a Dev Container to run commands for CI, testing, and more, along with pre-building Dev Container image. Dev Container image building supports [Dev Container Features](https://containers.dev/implementors/features/#devcontainer-json-properties) and automatically places Dev Container [metadata on an image](https://containers.dev/implementors/spec/#image-metadata) label for simplified use.
 
-This project builds on top of [@devcontainers/cli](https://www.npmjs.com/package/@devcontainers/cli)
+A similar [Azure DevOps Task](./docs/azure-devops-task.md) is also available!
 
+Note that this project builds on top of [@devcontainers/cli](https://www.npmjs.com/package/@devcontainers/cli) which can be used in other automation systems.
 
 ## GitHub Action
+The examples below show usage of the GitHub Action - see the [GitHub Action documentation](./docs/github-action.md) for more details.
 
-The example below shows usage of the GitHub Action - see the [GitHub Action documentation](./docs/github-action.md) for more details:
+> **NOTE:** This Action is not currently capable of taking advantage of pre-built Codespaces. However, pre-built images are supported.
 
 
+**Pre-building an image:**
+
 ```yaml
-- name: Build and run dev container task
+- name: Pre-build dev container image
   uses: devcontainers/ci@v0.2
   with:
     imageName: ghcr.io/example/example-devcontainer
-    runCmd: make ci-build
+    cacheFrom: ghcr.io/example/example-devcontainer
+    push: true
 ```
 
-## Azure DevOps Task
-
-The example below shows usage of the Azure DevOps Task - see the [Azure DevOps Task documentation](./docs/azure-devops-task.md) for more details:
+**Using a dev container for a CI build:**
 
 ```yaml
-- task: DevcontainersCI@0
-  inputs:
-    imageName: 'yourregistry.azurecr.io/example-dev-container'
-    runCmd: 'make ci-build'
-    sourceBranchFilterForPush: refs/heads/main
+- name: Run make ci-build in dev container
+  uses: devcontainers/ci@v0.2
+  with:    
+    # [Optional] Even if you're not directly referencing a pre-built image,
```

Comment:
I'm not sure I understand this example. This is referencing an image via `cacheFrom`, but what has created that image?

Reply:
I never use the same workflow to create the image and run CI.  I have one workflow that focuses on image generation, and then several different ones that use it to execute things within it.

Is there another way to do that?

Reply:
The comment in the example says:

```
# [Optional] Even if you're not directly referencing a pre-built image,
# you can still use the image as a cache for your Dockerfile build.
```

... which didn't conjure up the scenario you described. If I've understood correctly, this example is showing how you can reference an image that was built elsewhere and doesn't then push the image to a registry afterwards?

Reply:
And in terms of how I use the action, I often have a workflow with multiple jobs where the first one builds & pushes the image (and typically runs something in the container), and then other jobs in the workflow use the image as a cache and perform other steps.

Don't know whether that works in your case?

Reply:
Not in my case - different timing and triggers. 

I am frequently creating new repositories and at the start I have nothing to build - but I want a pre-built dev container image.  So, I add that first and trigger its build on a regular basis (to get patches), and whenever the `.devcontainer` folder is touched.

As I mature, I add other workflows with completely different triggers.

Reply:
Updated comment to make this clearer!

Reply:
Great - the update conjures a much clearer picture to me. Thanks!